python_utils_test.py

```diff
@@ -154,22 +154,65 @@ def test_url_join(self):
             'http://www.cwi.nl/%7Eguido/Python.html', 'FAQ.html')
         self.assertEqual(response, 'http://www.cwi.nl/%7Eguido/FAQ.html')
 
-    def test_recursively_convert_to_str(self):
+    def test_recursively_convert_to_str_with_dict(self):
         test_var_1 = python_utils.UNICODE('test_var_1')
         test_var_2 = python_utils.UNICODE('test_var_2')
         test_var_3 = test_var_1.encode('utf-8')
         test_var_4 = test_var_2.encode('utf-8')
         test_dict = {test_var_1: test_var_3, test_var_2: test_var_4}
+        self.assertEqual(
+            test_dict,
+            {'test_var_1': b'test_var_1', 'test_var_2': b'test_var_2'})
 
         for key, val in test_dict.items():
             self.assertEqual(type(key), future.types.newstr)
             self.assertEqual(type(val), future.types.newbytes)
 
-        dict_in_str = python_utils.recursively_convert_to_str(test_dict)
+        dict_in_str = python_utils._recursively_convert_to_str(test_dict)  # pylint: disable=protected-access
+        self.assertEqual(
+            dict_in_str,
+            {'test_var_1': 'test_var_1', 'test_var_2': 'test_var_2'})
+
         for key, val in dict_in_str.items():
             self.assertEqual(type(key), unicode)
             self.assertEqual(type(val), bytes)
 
+    def test_recursively_convert_to_str_with_list(self):
+        test_var_1 = python_utils.UNICODE('test_var_1')
+        test_list_1 = [
+            test_var_1, test_var_1.encode('utf-8'), 'test_var_2', b'test_var_3',
+            {'test_var_4': b'test_var_5'}]
+        test_list_2 = [{test_var_1: test_list_1}]
+        self.assertEqual(
+            test_list_2,
+            [{
+                'test_var_1': [
+                    'test_var_1', b'test_var_1', 'test_var_2', b'test_var_3',
+                    {'test_var_4': b'test_var_5'}]
+                }
+            ])
+
+        list_in_str = python_utils._recursively_convert_to_str(test_list_2)  # pylint: disable=protected-access
+        self.assertEqual(
+            test_list_2,
+            [{
+                'test_var_1': [
+                    'test_var_1', b'test_var_1', 'test_var_2', 'test_var_3',
+                    {'test_var_4': 'test_var_5'}]
+                }
+            ])
+
+        for item in list_in_str:
+            self.assertNotEqual(type(item), future.types.newstr)
+            self.assertNotEqual(type(item), future.types.newbytes)
+
+        dict_in_str = list_in_str[-1]
+        for key, value in dict_in_str.items():
+            self.assertNotEqual(type(key), future.types.newstr)
+            self.assertNotEqual(type(key), future.types.newbytes)
+            self.assertNotEqual(type(value), future.types.newstr)
+            self.assertNotEqual(type(value), future.types.newbytes)
```

Comment:
Add a self.assertEqual(type(key), ...) here too, ditto for value

Reply:
Done